theia-extensions/viewer-prototype/src/browser/trace-viewer/trace-viewer-frontend-module.ts

```diff
@@ -88,4 +93,28 @@ export default new ContainerModule(bind => {
             return connection.createProxy<BackendFileService>(backendFileServicePath);
         })
         .inSingletonScope();
+
+    bind(TspClient)
+        .toDynamicValue(ctx => {
+            const connection = ctx.container.get(WebSocketConnectionProvider);
+            return connection.createProxy<TspClient>(TRACE_SERVER_CLIENT);
+        })
+        .inSingletonScope();
+
+    bind(PortPreferenceProxy)
+        .toDynamicValue(ctx => {
+            const connection = ctx.container.get(WebSocketConnectionProvider);
+            return connection.createProxy<PortPreferenceProxy>(TRACE_SERVER_PORT);
+        })
+        .inSingletonScope();
+
+    bind(TraceServerConnectionStatusBackend)
+        .toDynamicValue(ctx => {
+            const connection = ctx.container.get(WebSocketConnectionProvider);
+            return connection.createProxy<TraceServerConnectionStatusBackend>(TRACE_SERVER_CONNECTION_STATUS);
+        })
+        .inSingletonScope();
+
+    bind(TspFrontendClientImpl).toSelf().inSingletonScope();
+    bind(TspFrontendClient).to(TspFrontendClientImpl).inSingletonScope();
```

Comment:
Perhaps simplify this binding to
```suggestion
    bind(TspFrontendClient).to(TspFrontendClientImpl).inSingletonScope();
```